src/main/java/org/wso2/carbon/module/swiftiso20022/ISO20022Camt053Validator.java

```diff
@@ -0,0 +1,116 @@
+
+        // Processing ISO20022.camt.053 message
+        try {
+            String xPath = ISOMessageParser.constructXPath(isBusinessMsg, ConnectorConstants.XPATH_DOCUMENT);
+            String documentStr = ISOMessageParser.extractISOMessage(messageContext, xPath);
+
+            XSDValidator documentValidator = new XSDValidator(ConnectorConstants.XSD_SCHEMA_CAMT_053_001);
+            documentValidator.validateXMLContent(documentStr);
+
+            // Validate MT940 related ISO20022.camt.053 validations
+            if (!ISO20022Camt053ValidatorUtils.isElectronicSequenceNumberExists(isBusinessMsg, messageContext)) {
+                this.log.error(ConnectorConstants.ERROR_MISSING_ELECTRONIC_SEQUENCE_NUMBER);
+                ConnectorUtils.appendErrorToMessageContext(messageContext,
+                        ConnectorConstants.ERROR_INVALID_ISO_CAMT053_XML_MSG,
+                        ConnectorConstants.ERROR_MISSING_ELECTRONIC_SEQUENCE_NUMBER);
+
+                throw new ConnectException(ConnectorConstants.ERROR_MISSING_ELECTRONIC_SEQUENCE_NUMBER);
+            } else if (!ISO20022Camt053ValidatorUtils.isLegalSequenceNumberExists(isBusinessMsg, messageContext)) {
+                this.log.error(ConnectorConstants.ERROR_MISSING_LEGAL_SEQUENCE_NUMBER);
+                ConnectorUtils.appendErrorToMessageContext(messageContext,
+                        ConnectorConstants.ERROR_INVALID_ISO_CAMT053_XML_MSG,
+                        ConnectorConstants.ERROR_MISSING_LEGAL_SEQUENCE_NUMBER);
+
+                throw new ConnectException(ConnectorConstants.ERROR_MISSING_LEGAL_SEQUENCE_NUMBER);
+            } else if (!ISO20022Camt053ValidatorUtils.isOpeningBalanceExists(isBusinessMsg, messageContext)) {
+                this.log.error(ConnectorConstants.ERROR_MISSING_OPENING_BALANCE);
+                ConnectorUtils.appendErrorToMessageContext(messageContext,
+                        ConnectorConstants.ERROR_INVALID_ISO_CAMT053_XML_MSG,
+                        ConnectorConstants.ERROR_MISSING_OPENING_BALANCE);
+
+                throw new ConnectException(ConnectorConstants.ERROR_MISSING_OPENING_BALANCE);
+            } else if (!ISO20022Camt053ValidatorUtils.isClosingBalanceExists(isBusinessMsg, messageContext)) {
+                this.log.error(ConnectorConstants.ERROR_MISSING_CLOSING_BALANCE);
+                ConnectorUtils.appendErrorToMessageContext(messageContext,
+                        ConnectorConstants.ERROR_INVALID_ISO_CAMT053_XML_MSG,
+                        ConnectorConstants.ERROR_MISSING_CLOSING_BALANCE);
+
+                throw new ConnectException(ConnectorConstants.ERROR_MISSING_CLOSING_BALANCE);
+            }
+
+            this.log.debug("Valid camt.053.001.11 message");
+        } catch (SAXParseException e) {
+            String errMsg = String.format("%s, Line number: %d, Column number: %d",
+                    e.getMessage(), e.getLineNumber(), e.getColumnNumber());
+
+            this.log.error(e.getMessage());
```

Comment:
Lets log e as well

Reply:
Check all places

Reply:
Added in a335dc1f39d3a52600c391d8b6daf8093be6f42c